c7n/resources/iam.py

```diff
@@ -774,6 +774,120 @@ def _user_mfa_devices(resource):
         return matched
 
 
+@User.filter_registry.register('username')
+class UserName(ValueFilter):
```

Comment:
you're actually completely right. the following snippet acchieves the same, without adding a new filter:

```
policies:
  - name: defined-machine-users
    resource: iam-user
    filters:
      - type: value
        key: "Arn"
        op: not-in
        value:
          - arn:aws:iam::123456789012:user/valid-user-1
          - arn:aws:iam::123456789012:user/valid-user-2
```